Wrappers/Python/cil/optimisation/operators/GradientOperator.py

```diff
@@ -261,45 +274,53 @@ class Gradient_C(LinearOperator):
                      on 2D, 3D, 4D ImageData
                      under Neumann/Periodic boundary conditions'''
 
-    def __init__(self, gm_domain, gm_range=None, bnd_cond = NEUMANN, **kwargs):
+    def __init__(self, domain_geometry,  bnd_cond = NEUMANN, **kwargs):
 
+        # Number of threads
         self.num_threads = kwargs.get('num_threads',NUM_THREADS)
+        
+        # Split gradients, e.g., space and channels
         self.split = kwargs.get('split',False)
-        self.gm_domain = gm_domain
-        self.gm_range = gm_range
-        self.ndim = self.gm_domain.length
-                
+        
+        # Consider pseudo 3D geometries with one slice, e.g., (1,155,155) for SIRF data
+        self.is2D = False
+        self.tmp_dom_shape = []
```

Comment:
this variable deserves a better name. It is the domain shape after we remove axis of length 1. I believe calling it `self.domain_shape` would make it clearer.